### ISSUE TITLE
[turborepo] turbo-ignore always builds all projects for new branches on Vercel

### ISSUE THREAD

**Issue by kieranm** [vercel/turbo#6310]
### What version of Turborepo are you using?

turbo-ignore@1.10.16

### What package manager are you using / does the bug impact?

pnpm

### What operating system are you using?

Mac

### Describe the Bug

We have a monorepo with multiple websites, and for each Vercel app we have set the ignored build step to `npx turbo-ignore`. We have the Git integration enabled for preview builds.

When we open a PR, we expect that only changed apps will be built, to reduce unnecessary builds.

However, in Vercel I see the following output:

```
Running "npx turbo-ignore"
--
22:51:51.448 | npm WARN exec The following package was not found and will be installed: turbo-ignore@1.10.16
22:51:51.767 | ≫   Using Turborepo to determine if this project is affected by the commit...
22:51:51.768 |  
22:51:51.772 | ≫   Inferred "@daybridge/daybridge.design" as workspace from "package.json"
22:51:51.772 | ≫   Using "build" as the task from the arguments
22:51:51.773 | ≫   No previous deployments found for "@daybridge/daybridge.design" on branch "day-2084-item-form-does-not-appear-for-items-ending-at-midnight-on"
22:51:51.773 | ✓ Proceeding with deployment
```

The `@daybridge/daybridge.design` app did not have any changed files, so it should have been ignored. But the message suggests that all web apps are being built because they haven't been deployed before on this branch.

### Expected Behavior

I'm not sure if I'm doing something wrong, or if there is some context I'm missing, but it's important that apps that have not been changed don't deploy unnecessarily, so I would expect turbo-ignore to be skipping these even if they haven't been deployed before for a specific branch.

### To Reproduce

- Monorepo with one or more web projects
- Each project added as a separate app to Vercel, with preview builds enabled
- Each web project has its ignored build step set to `npx turbo-ignore`
- Make a change impacting only one of the web projects and none of the others
- Open PR in Github

**Comment by gsoltis** [comment]
Can you try setting `--fallback` to handle scenarios when it's the first deploy of a particular branch?

See [Usage](https://www.npmjs.com/package/turbo-ignore#usage)

**Comment by kieranm** [comment]
Thanks! I can definitely try setting this, but the guidance on Vercel's docs says its only necessary to use `npx turbo-ignore` without any additional flags. This used to work just fine. So I would like to understand why this has started happening, and potentially whether its a bug.

**Comment by kieranm** [comment]
In fact, in Vercel this field is actually disabled so it's not possible to change it without bailing out and setting a custom command.

<img width="955" alt="Screenshot 2023-10-30 at 10 19 44" src="https://github.com/vercel/turbo/assets/232283/d157584f-3b9d-4098-a6d8-a71b65533770">

**Comment by gsoltis** [comment]
Is this only for the first commit of a new branch, or is this happening on every subsequent commit as well?

**Comment by kieranm** [comment]
Thanks for the explanation! The behaviour makes sense

If there are no previous deploys for a branch, is there a way of checking for changes by comparing with the last deploy of the branch you are merging **into**? Eg. if `feat-1` was being merged into `staging`, look at the last successfully deployed `staging` commit and then compare against that?

In any case, I do think the copy on Vercel needs to change or be clarified, because it's currently a bit misleading if apps are going to be deployed for safety even if they haven't changed vs the base branch. The behaviour is sensible but just needs to be made clearer I think.

**Comment by tknickman** [comment]
Yes that would be the best solution here, and we're working on something intenrally to solve this that should be available soon. 

Right now, this is a limitation on Vercel due to the way the repo is cloned (to keep clones speedy, we only get details about the current branch, no others) so Vercel doesn't know anything about the base to compare against.
